config/dr-cluster/default/kustomization.yaml

```diff
@@ -11,6 +11,21 @@ namePrefix: ramen-dr-cluster-
 # Labels to add to all resources and selectors.
 #commonLabels:
 #  someName: someValue
+transformers:
+- |-
+  apiVersion: builtin
+  kind: LabelTransformer
+  metadata:
+    name: temporary
+  labels:
+    app: ramen-dr-cluster
+  fieldSpecs:
+  - kind: Deployment
+    path: spec/selector/matchLabels
+  - kind: Deployment
+    path: spec/template/metadata/labels
+  - kind: Service
+    path: spec/selector
```

Comment:
Add the following here:
```
  - kind: Namespace
    path: metadata/labels
  - kind: Service
    path: metadata/labels
  - kind: Deployment
    path: metadata/labels
```

Just covering resources that are currently labelled as well, so that they can be grouped/found with the label.

The namespace label will get overwritten if we deploy to the same cluster (as the key is the same), so maybe we drop that from the list above.

(applies to the hub kustomize files too, if we adopt it)

Reply:
I experimented with `kustomize`'s `commonLabels` which applies the label in many more places:

> Labels transformer
The labels transformer adds labels to the metadata/labels field for all resources. It also adds labels to the spec/selector field in all Service resources as well as the spec/selector/matchLabels field in all Deployment resources.

https://github.com/kubernetes-sigs/kustomize/blob/master/examples/transformerconfigs/README.md#labels-transformer

In addition to the labels already included, `commonLabels` would apply the label to the metadata/labels of the following kinds for both the `hub` and `dr-cluster`:

```
hub:
Namespaces
CustomResourceDefinitions
ServiceAccounts
Roles
ClusterRoles
RoleBindings
ClusterRoleBindings
ConfigMaps
Services
Deployments
```

It seems the label is not desired for all of these kinds.  I chose to use the `LabelTransformer` method as it provides a way to whitelist the paths to label.  It seems like this is still the right approach.

The point about the namespace is a good one.  Perhaps different namespaces for the `hub` and `dr-cluster` are desirable; however, I understand the namespace name may be specified by a user who chooses to put them in the same namespace on the same cluster.

Reply:
For now, I'll just add the label to `Deployment` and `Service` metadata/label as you suggested.

Reply:
Ack! also, agree on `commonLabels`, hence chose to just label the deployment and the service.